services/src/model_training/xgboost_training.rs

```diff
@@ -0,0 +1,127 @@
+use std::collections::hash_map::Entry;
+use std::{collections::HashMap, path::PathBuf};
+
+use serde::{Deserialize, Serialize};
+use typetag::serde;
+
+use serde_json::Value;
+use snafu::{ensure, ResultExt};
+use xgboost_rs::{Booster, DMatrix};
+
+use crate::error::Result;
+use crate::model_training::xg_error as XgModuleError;
+use geoengine_datatypes::pro::MachineLearningFeature;
+use ordered_float::NotNan;
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct XgboostTrainingParams {
+    // where to store the model file
+    pub model_store_path: Option<PathBuf>,
```

Comment:
this shouldn't be controlled by the users but rather in the Settings.toml

Reply:
I've changed the xgboost prediction operator file now such that the model path is reconstructed by uuid.
But I think that this is now subject to a more deliberate change, since this module was written when we did not have the model training routine yet.
Now it's still using the `ml_models` hashmap, where the model content is stored temporarily.
I guess it's now time to change all that to use only the uuid and the config settings, isn't it?

046ebded

commit msg:
refactor: xgboost prediction operator now reads a trained model by uuid path (WIP)

Reply:
We should use the UUID (or better create a new Id-Type using `identifer!`: `ModelId`), to identify the models instead of a path, yes.

Regarding the `ml_models` hashmap: Why don't we just use it? Can the `MachineLearningModelFromWorkflowTask` not use the `write_ml_model` to store the resulting model instead of writing it to disk itself. Then the execution context can do the specifics (later e.g. writing to postgres). And the XgBoost(Prediction)Operator can use the `read_ml_model` to get it back again using the model_id.

For this to work, the read/write_ml_model methods should operate on the `ModelId`, instead of `Path`s

Reply:
implemented in 
2cdc485

---
commit msg:
refactor: implement a `ModelId` that is used in conjunction with ml models